lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -425,6 +464,12 @@ def ensure(module, base, state, names):
             # packages
             allow_erasing = True
 
+            if autoremove:
+                if LooseVersion(dnf.__version__) < LooseVersion('2.0.1'):
+                    module.fail_json(msg="Autoremove requires dnf>=2.0.1. Current dnf version is %s" % dnf.__version__)
+                else:
```

Comment:
We should do this autoremove + dnf.__version__ check in main() soon after we create our AnsibleModule.  That way if there's a problem, the user doesn't have to wait for any other processing to occur before being told to change the parameters they're giving to the module.